servicetalk-encoding-api/src/main/java/io/servicetalk/encoding/api/AbstractZipContentCodec.java

```diff
@@ -138,11 +138,18 @@ public void onNext(Buffer next) {
                         // +1 for the encoding footer (ie. END_OF_STREAM)
                         try {
                             if (next == END_OF_STREAM) {
-                                // ZIP footer is 10 bytes
-                                Buffer dst = allocator.newBuffer(10);
-                                stream.swap(dst);
-                                output.finish();
+                                Buffer dst;
+                                if (!headerWritten) {
+                                    // Empty publisher will not produce any items other than the END_OF_STREAM
+                                    // so we need to keep the header & footer in the same chunk and deliver downstream.
+                                    dst = stream.buffer;
```

Comment:
is there any value to pre-allocating the `Buffer` now that each `onNext` checks if `headerWritten` and potentially does an allocation? consider just allocating the buffer on demand in each onNext and swapping to reduce conditional blocks.